### PR TITLE
Implement --output flag, add command; get projects, refactor testing codes

### PR DESCRIPTION
**Pull request by bzon**
Fixes https://github.com/devopsctl/gitlabctl/issues/4 and https://github.com/devopsctl/gitlabctl/issues/56

### CODE REVIEW

File: cmd/flag_helpers.go [hunk 1]
```diff
@@ -30,34 +31,98 @@ import (
...
-		"Order groups by name or path. Default is name")
+	addAllAvailableFlag(cmd)
+	addGroupOrderByFlag(cmd)
+	addOwnedFlag(cmd)
+	addSortFlag(cmd)
+	addStatisticsFlag(cmd)
+	addSearchFlag(cmd)
+}
+
+// addGetProjectsFlags adds common flags for `get projects` commands
```
Comment (depwaswho94) [inline cmd/flag_helpers.go]: The default sort in projects api is desc but in this function it is asc and in the flag description seems fixed for groups ?

File: cmd/flag_helpers.go [hunk 2]
```diff
@@ -30,34 +31,98 @@ import (
...
+func addGetProjectsFlags(cmd *cobra.Command) {
+	cmd.Flags().Bool("archived", false,
+		"Limit by archived status")
+	addProjectOrderByFlag(cmd)
+	addSortFlag(cmd)
```
Comment (depwaswho94) [inline cmd/flag_helpers.go]: The default sort in projects api is desc but in this function it is asc and in the flag description seems fixed for groups ?

Comment (bzon) [reply cmd/flag_helpers.go]: Fixing the description in the incoming commit. 

Let us settle with the `asc` as default to have a common `addSortFlag` and avoid code duplication. The impact is not that important, I think.

Comment (depwaswho94) [reply cmd/flag_helpers.go]: Ok sir agree with that.

**Pull request merged**